### PR TITLE
Hook into the oembed template to inject a buy button for tickets

### PR DESCRIPTION
**Pull request by borkweb**
Related: https://github.com/moderntribe/the-events-calendar/pull/647

See: https://central.tri.be/issues/43938

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$buy_button' | [Main.php:479](https://github.com/moderntribe/event-tickets/blob/42fbe08c0cb8a0c5ccfa5a69212cd23e6daa1777/src/Tribe/Main.php#L479)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -417,4 +419,77 @@ public function inject_post_types( $post_types ) {
...
+		<?php
+		$buy_button = ob_get_clean();
+
+		/**
+		 * Filters the buy button that appears on event oembeds
+		 *
+		 * @var string The button markup
+		 * @var int Event ID
+		 */
+		$buy_button = apply_filters( 'event_tickets_embed_buy_button', $buy_button, $event_id );
```
Comment (barryhughes) [inline src/Tribe/Main.php]: Echo directly instead of allocating to a var?

Comment (borkweb) [inline src/Tribe/Main.php]: I tend to like to allocate to a variable because it makes debugging easier later on. But I can change it.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Main.php:477](https://github.com/moderntribe/event-tickets/blob/03b02df0b783a34edcc8389e234be7e946b4fc34/src/Tribe/Main.php#L477)

> via jenkins/codesniffer-phpcs

**Comment by barryhughes** [comment]
:+1:

**Pull request merged**